app/services/UpdateIncomeSourceService.scala

```diff
@@ -44,4 +45,18 @@ class UpdateIncomeSourceService @Inject()(connector: IncomeTaxViewChangeConnecto
 
   }
 
+  def updateCessationDatev2(nino: String, incomeSourceId: String, cessationDate: String)
+                           (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[UpdateIncomeSourceError.type, UpdateIncomeSourceSuccess.type]]= {
+    connector.updateCessationDate(
```

Comment:
I would suggest to change next case classes signature for UpdateIncomeSourceError and  UpdateIncomeSourceSuccess to:

`
case class UpdateIncomeSourceError(reson: String)
case class UpdateIncomeSourceSuccess(incomeSourceId: String)
// and also method signature:
  def updateCessationDatev2(nino: String, incomeSourceId: String, cessationDate: String)
                           (implicit hc: HeaderCarrier, ec: ExecutionContext): **Future[Either[UpdateIncomeSourceError, UpdateIncomeSourceSuccess]]**
`

lets assume that caller of this service method (controller) would decided if we need to log these details or not